include/ginkgo/core/preconditioner/jacobi.hpp

```diff
@@ -371,6 +371,13 @@ class Jacobi : public EnableLinOp<Jacobi<ValueType, IndexType>>,
         gko::array<index_type> GKO_FACTORY_PARAMETER_VECTOR(block_pointers,
                                                             nullptr);
 
+        /**
+         * Use L1 Jacboi.
+         * If it is true, it generates the preconditioner on A + Diag(sum_{k in
+         * off diagonal block of i} A_ik) not A.
+         */
+        bool GKO_FACTORY_PARAMETER_SCALAR(l1, false);
```

Comment:
I'm still not convinced that `l1` is a good name. Perhaps `aggregate`, or `aggregate_l1`.
One issue with just `l1` is that we will also have the distributed L1 preconditioner, as mentioned in the original paper. But that is quite different from the one here, so we need better naming to prevent confusion.

Reply:
I think they are the same thing. They only use different way to define block  (by node, diagonal block, or value), but they all add the L1 norm from the row in off-block-diag to the diag.

Reply:
They are not the same. Using the l1 defined here will not give you a distributed preconditioner. Also the distributed version can be used both with local Jacobi and Gauss-Seidel preconditioner.

Reply:
I do not get it.
If you have the block list, the diagonal block { $B_i$ } and the off diagonal block { $O_i$ }
the L1 does the diagonal value update
the diagonal value $d_k$ in $B_i$ is updated by $d_k$ += sum abs( value in the same row as $d_k$ in $O_i$ )

For sure, L1 in this pr is specific for Jacobi because we can only know the block of block Jacobi in generation.
scalar L1 Jacobi takes each diagonal value as the block
block L1 Jacobi takes the blocks generated or given as the block
distributed L1 takes the local matrix (assume squared) as the block.

The diagonal treatment are the same

Reply:
The point is this is a variation of the sequential jacobi. In the distributed case, we would have a separate L1 preconditioner (or with a different name) that is not specific to jacobi.
Having both with the same name would be confusing.

Reply:
I see